File: streaming-service/src/main/java/io/spring/demo/streaming/StreamingServiceApplication.java
package io.spring.demo.streaming;


import java.time.Duration;
import java.util.Arrays;
import java.util.List;

import com.samskivert.mustache.Mustache;
import io.spring.demo.streaming.portfolio.User;
import io.spring.demo.streaming.portfolio.UserRepository;
import io.spring.demo.streaming.support.MustacheResourceTemplateLoader;
import io.spring.demo.streaming.support.MustacheViewResolver;

import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.mustache.MustacheProperties;
import org.springframework.context.annotation.Bean;

@SpringBootApplication
public class StreamingServiceApplication {

	public static void main(String[] args) {
		SpringApplication.run(StreamingServiceApplication.class, args);
	}

	@Bean
	public MustacheViewResolver mustacheViewResolver(MustacheProperties properties) {
		MustacheViewResolver viewResolver = new MustacheViewResolver();
		viewResolver.setPrefix(properties.getPrefix());
		viewResolver.setSuffix(properties.getSuffix());
		MustacheResourceTemplateLoader loader =
				new MustacheResourceTemplateLoader(properties.getPrefix(), properties.getSuffix());
		viewResolver.setCompiler(Mustache.compiler().escapeHTML(false).withLoader(loader));
		return viewResolver;
	}

	@Bean
	public CommandLineRunner createUsers(UserRepository userRepository) {
		return strings -> {
			List<User> users = Arrays.asList(
					new User("sdeleuze", "Sebastien Deleuze"),
					new User("snicoll", "Stephane Nicoll"),
					new User("rstoyanchev", "Rossen Stoyanchev"),
					new User("smaldini", "Stephane Maldini"),
					new User("sbasle", "Simon Basle"),
					new User("bclozel", "Brian Clozel")
			);
			userRepository.save(users).blockLast(Duration.ofSeconds(3));
		};
	}

}

File: streaming-service/src/main/kotlin/io/spring/demo/streaming/StreamingServiceApplication.kt
package io.spring.demo.streaming;

import io.spring.demo.streaming.portfolio.User
import io.spring.demo.streaming.portfolio.UserRepository
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import java.time.Duration.ofSeconds

@SpringBootApplication
class StreamingServiceApplication {

	@Bean
	fun createUsers(userRepository: UserRepository) = CommandLineRunner {
		val users = listOf(
				User("sdeleuze", "Sebastien Deleuze"),
				User("snicoll", "Stephane Nicoll"),
				User("rstoyanchev", "Rossen Stoyanchev"),
				User("smaldini", "Stephane Maldini"),
				User("simonbasle", "Simon Basle"),
				User("bclozel", "Brian Clozel")
		)
		userRepository.saveAll(users).blockLast(ofSeconds(3))
	}

}

fun main(args: Array<String>) {
	SpringApplication.run(StreamingServiceApplication::class.java, *args)
}
